lib/workers/repository/onboarding/branch/rebase.ts

```diff
@@ -0,0 +1,60 @@
+import { GlobalConfig } from '../../../../config/global';
+import type { RenovateConfig } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { scm } from '../../../../modules/platform/scm';
+import { toSha256 } from '../../../../util/hash';
+import { defaultConfigFile } from '../common';
+import { OnboardingCommitMessageFactory } from './commit-message';
+import { getOnboardingConfigContents } from './config';
+
+export async function rebaseOnboardingBranch(
+  config: RenovateConfig,
+  previousConfigHash: string | undefined
+): Promise<string | null> {
+  logger.debug('Checking if onboarding branch needs rebasing');
+
+  // skip platforms that do not support html comments in pr
+  const pl = GlobalConfig.get('platform')!;
+  if (!['github', 'gitea', 'gitlab'].includes(pl)) {
+    return null;
+  }
+
+  const configFile = defaultConfigFile(config);
+  const contents = await getOnboardingConfigContents(config, configFile);
+  const currentConfigHash = toSha256(contents);
+
+  if (previousConfigHash === currentConfigHash) {
+    logger.debug('No rebase needed');
+    return null;
+  }
+  logger.debug(
+    { previousConfigHash, currentConfigHash },
+    'Rebasing onboarding branch'
+  );
+
+  const commitMessageFactory = new OnboardingCommitMessageFactory(
+    config,
+    configFile
+  );
+  const commitMessage = commitMessageFactory.create();
+
+  if (GlobalConfig.get('dryRun')) {
+    logger.info('DRY-RUN: Would rebase files in onboarding branch');
+    return null;
+  }
```

Comment:
I think we can save a litle time when doing this before commit message initializing. 🤔